crates/rover-graphql/src/lib.rs

```diff
@@ -189,7 +210,37 @@ where
                 })?;
             if let Some(errors) = graphql_response.errors {
                 match graphql_response.data {
-                    Some(data) => Err(GraphQLServiceError::PartialError { data, errors }),
+                    Some(data) => {
+                        let mut friendly_errors_detail: Vec<String> = vec![];
+                        for err in errors.iter() {
+                            let json_root =
+                                err.extensions.as_ref().unwrap().get("response").unwrap();
+                            friendly_errors_detail = serde_json::from_str::<PartialErrorInnerBody>(
+                                &json_root.to_string(),
+                            )
+                            .unwrap()
+                            .body
+                            .errors
+                            .iter()
+                            .fold(vec![], |mut acc, x| {
+                                acc.push(x.clone().message);
+                                acc
+                            });
+                        }
+
+                        if friendly_errors_detail
+                            .join("")
+                            .contains("Invalid credentials")
+                        {
+                            Err(GraphQLServiceError::InvalidCredentials {})
+                        } else {
+                            Err(GraphQLServiceError::PartialError {
+                                data,
+                                errors,
+                                friendly_errors_detail,
+                            })
+                        }
+                    }
```

Comment:
I really like this approach, I wonder if we can turn the screw with Serde even more, can we use it to deserialise the whole response and turn the errors into enums? Then you could just match on those without having to search and do string matching?